extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -280,19 +281,18 @@ class MongoQueueProcessor {
                 dataStoreETag: `1:${emptyFileMd5}`,
                 dataStoreVersionId,
             }];
-            entry.setLocation(editLocation);
         } else {
-            const editLocations = locations.map(location => {
-                const newValues = {
-                    key: entry.getObjectKey(),
-                    dataStoreName: zenkoLocation,
-                    dataStoreType: 'aws_s3',
-                    dataStoreVersionId,
-                };
-                return Object.assign({}, location, newValues);
-            });
-            entry.setLocation(editLocations);
+            zenkoDataLocations = [{
+                key: entry.getObjectKey(),
+                size: entry.getContentLength(),
+                start: 0,
+                dataStoreName: zenkoLocation,
+                dataStoreType: 'aws_s3',
+                dataStoreETag: entry.getContentMd5(),
```

Comment:
Shouldn't it be `1:${entry.getContentMd5()}`?

Reply:
True.

Reply:
Just checked `ObjectMDLocation` model. You are right.

Reply:
what about actually using the model :wink: not a blocker though

Reply:
A refactor is coming 🔜